### PR TITLE
Use ts-tinkerpop@1.6, java@0.6.0, compat w/ node v4

### PR DESCRIPTION
**Pull request by jimlloyd**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,6 +45,6 @@
     "debug": "^2.2.0",
     "graphviz": "0.0.8",
     "lodash": "^3.10.1",
-    "ts-tinkerpop": "=1.5.1"
+    "ts-tinkerpop": "^1.6.0"
```
Comment (mhfrantz) [inline package.json]: Don't we want to pin the version?

**Comment by mhfrantz** [comment]
FTP

File: package.json (hunk 1 above)
Comment (jimlloyd) [inline package.json]: Yeah, thanks, I should have noticed that.

**Comment by jimlloyd** [comment]
PTAL

**Comment by mhfrantz** [comment]
LGTM pending CI

**Pull request merged**